core/src/main/java/org/apache/brooklyn/core/objs/AbstractBrooklynObject.java

```diff
@@ -55,8 +57,18 @@
     @SetFromFlag(value = "tags")
     private final Set<Object> tags = Sets.newLinkedHashSet();
 
+    private RelationSupportInternal<PublicSelfType> relations = new ByObjectBasicRelationSupport<PublicSelfType>(getPublicThis(), new RelationChangedCallback());
```

Comment:
Went to investigate a bit deeper as it didn't seem right. The constructor of `BrooklynObject`s is called because we are asking Brooklyn to instantiate the object. But for plain java objects which happen to be config/sensor values xstream handles instantiation and it won't call any of the constructors.